strategies/affected/src/test/java/org/arquillian/smart/testing/strategies/affected/ClassDependenciesGraphUsingPropertyTest.java

```diff
@@ -0,0 +1,121 @@
+
+        final Configuration configuration = Configuration.load();
+        configuration.setStrategiesConfiguration(Collections.singletonList(affectedConfiguration));
+        configuration.dump(temporaryFolder.getRoot());
+
+        final ClassDependenciesGraph
+            classDependenciesGraph = new ClassDependenciesGraph(new EndingWithTestTestVerifier(), Configuration.load());
+
+        final String testLocation = ATest.class.getResource("ATest.class").getPath();
+        final String testLocation2 = BTest.class.getResource("BTest.class").getPath();
+        final String testLocation3 = CTest.class.getResource("CTest.class").getPath();
+        classDependenciesGraph.buildTestDependencyGraph(Arrays.asList(new File(testLocation), new File(testLocation2),
+            new File(testLocation3)));
+
+        // when
+        Set<File> mainObjectsChanged = new HashSet<>();
+        mainObjectsChanged.add(new File(D.class.getResource("D.class").getPath()));
+
+        final Set<String> testsDependingOn = classDependenciesGraph.findTestsDependingOn(mainObjectsChanged);
+
+        // then
+        assertThat(testsDependingOn)
+            .isEmpty();
+    }
+
+    @Test
+    public void should_exclude_imports_if_property_set() {
+        // given
+        final AffectedConfiguration affectedConfiguration = new AffectedConfiguration();
+        affectedConfiguration.setExclusions("org.arquillian.smart.testing.strategies.affected.fakeproject.main.B");
+
+        final Configuration configuration = Configuration.load();
+        configuration.setStrategiesConfiguration(Collections.singletonList(affectedConfiguration));
+        configuration.dump(temporaryFolder.getRoot());
+
+        final ClassDependenciesGraph
+            classDependenciesGraph = new ClassDependenciesGraph(new EndingWithTestTestVerifier(), Configuration.load());
+
+        final String testLocation = ATest.class.getResource("ATest.class").getPath();
+        final String testLocation2 = BTest.class.getResource("BTest.class").getPath();
+        final String testLocation3 = CTest.class.getResource("CTest.class").getPath();
+        classDependenciesGraph.buildTestDependencyGraph(Arrays.asList(new File(testLocation), new File(testLocation2),
+            new File(testLocation3)));
+
+        // when
+        Set<File> mainObjectsChanged = new HashSet<>();
+        mainObjectsChanged.add(new File(D.class.getResource("D.class").getPath()));
+
+        final Set<String> testsDependingOn = classDependenciesGraph.findTestsDependingOn(mainObjectsChanged);
+
+        // then
+        assertThat(testsDependingOn)
+            .isEmpty();
+    }
+
+    @Test
+    public void should_include_only_imports_if_property_set() {
+        // given
+        final AffectedConfiguration affectedConfiguration = new AffectedConfiguration();
+        affectedConfiguration.setInclusions("org.arquillian.smart.testing.strategies.affected.fakeproject.main.A");
```

Comment:
I've been trying to break this test anyhow - putting non-existing FQNs here, and the test is always green